Libraries/LLM/Load.swift

```diff
@@ -11,82 +11,91 @@ struct LLMError: Error {
     let message: String
 }
 
-/// Load and return the model and tokenizer
-public func load(
-    hub: HubApi = HubApi(), configuration: ModelConfiguration,
-    progressHandler: @Sendable @escaping (Progress) -> Void = { _ in }
-) async throws -> (LLMModel, Tokenizer) {
+func prepareModelDirectory(
+    hub: HubApi, configuration: ModelConfiguration,
+    progressHandler: @Sendable @escaping (Progress) -> Void
+) async throws -> URL {
     do {
-        let tokenizer = try await loadTokenizer(configuration: configuration, hub: hub)
-
-        let modelDirectory: URL
-
         switch configuration.id {
         case .id(let id):
             // download the model weights
             let repo = Hub.Repo(id: id)
             let modelFiles = ["*.safetensors"]
-            modelDirectory = try await hub.snapshot(
+            return try await hub.snapshot(
                 from: repo, matching: modelFiles, progressHandler: progressHandler)
 
         case .directory(let directory):
-            modelDirectory = directory
+            return directory
         }
+    } catch Hub.HubClientError.authorizationRequired {
+        // an authorizationRequired means (typically) that the named repo doesn't exist on
+        // on the server so retry with local only configuration
+        return configuration.modelDirectory(hub: hub)
+    }
+}
 
-        // create the model (no weights loaded)
-        let configurationURL = modelDirectory.appending(component: "config.json")
-        let baseConfig = try JSONDecoder().decode(
-            BaseConfiguration.self, from: Data(contentsOf: configurationURL))
+/// Load and return the model and tokenizer
+public func load(
+    hub: HubApi = HubApi(), configuration: ModelConfiguration,
+    progressHandler: @Sendable @escaping (Progress) -> Void = { _ in }
+) async throws -> (LLMModel, Tokenizer) {
```

Comment:
This is the original public method -- we keep it but the inside is refactored so that we can use the same code in the `actor` case.